pkg/mover/chart.go

```diff
@@ -255,13 +256,55 @@ func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
 
 // loadChartFromPath load the chart in memory from a given path
 func (cm *ChartMover) loadChartFromPath(path string) error {
-	var err error
-	if cm.chart, err = loader.Load(path); err != nil {
-		return &ChartLoadingError{Path: path, Inner: err}
+	// always unpack the chart tarball to a directory to avoid issues loading
+	// files if the tgz contains replicated entries
+	dir := path
+	info, err := os.Stat(path)
+	if err != nil {
+		return &ChartLoadingError{Path: dir, Inner: err}
+	}
+	if !info.IsDir() {
+		tmpdir, err := os.MkdirTemp("", "expanded-chart-*")
+		if err != nil {
+			return fmt.Errorf("failed to make temporary directory: %w", err)
+		}
+		defer os.RemoveAll(tmpdir)
+		if err != nil {
+			return fmt.Errorf("failed to detect the chart directory name: %w", err)
+		}
+		if err := untarz(true, path, "*", tmpdir); err != nil {
+			return fmt.Errorf("failed to extract chart: %w", err)
+		}
+		chartname, err := detectChartName(tmpdir)
+		if err != nil {
+			return fmt.Errorf("failed to detect chartname from extracted chart: %w", err)
+		}
+		dir = filepath.Join(tmpdir, chartname)
```

Comment:
is there any way to implement the equivalent to strip components?

```
$ tar xzf wordpress-12.2.4.tgz --strip-components=1
$ ls
Chart.lock  charts  Chart.yaml  ci  README.md  templates  values.schema.json  values.yaml  wordpress-12.2.4.tgz
```

Reply:
we agreed offline not to go there